common/djangoapps/enrollment/tests/test_api.py

```diff
@@ -230,3 +230,52 @@ def test_caching(self):
         # The data matches
         self.assertEqual(len(details['course_modes']), 3)
         self.assertEqual(details, cached_details)
+
+    def test_update_enrollment_expired_mode_with_error(self):
+        """ Verify that if verified mode is expired then enrollment cannot be updated if
+        include_expired flag is false."""
+        self.assert_add_modes_with_enrollment('audit')
+        # On updating enrollment mode to verified it should the raise the error.
+        with self.assertRaises(CourseModeNotFoundError):
+            self.assert_update_enrollment(mode='verified', include_expired=False)
+
+    def test_update_enrollment_with_expired_mode(self):
+        """ Verify that if verified mode is expired then enrollment can be updated if
+        include_expired flag is true."""
+        self.assert_add_modes_with_enrollment('audit')
+        # enrollment in verified mode will work fine with include_expired=True
+        self.assert_update_enrollment(mode='verified', include_expired=True)
+
+    @ddt.data(True, False)
+    def test_unenroll_with_expired_mode(self, include_expired):
+        """ Verify that un-enroll will work fine for expired courses whether include_expired
+        is true or false."""
+        self.assert_add_modes_with_enrollment('verified')
+        self.assert_update_enrollment(mode='verified', is_active=False, include_expired=include_expired)
+
+    def assert_add_modes_with_enrollment(self, enrollment_mode):
+        """ Dry method for adding fake course enrollment information to
+        the fake data API. Enroll the student in a course. """
```

Comment:
Dry method for adding fake course enrollment information to fake data API and enroll the student in the course.